scipy/stats/_sensitivity_analysis.py

```diff
@@ -542,28 +590,34 @@ def sobol_indices(
         A, B = sample_A_B(n=n, dists=dists, random_state=random_state)
         AB = sample_AB(A=A, B=B)
 
-        f_A = np.asarray(func(A))
+        f_A = np.atleast_2d(func(A))
 
-        if f_A.shape[0] != n or f_A.shape[1] < 1:
+        if f_A.shape[1] != n or f_A.shape[0] < 1:
```

Comment:
Is ` f_A.shape[0] < 1` possible now that you're using `np.atleast_2d`?

Reply:
Yes e.g. Ishigami is returning `array_like (n,)` now.

Reply:
I still don't understand. It seems unlikely that the zeroth axis of a 2d array will have size zero, even if it's empty.
```
np.atleast_2d([]).shape  # (1, 0)
```

Reply:
Ah ok I got your question now sorry. Yes looks like I can remove the second check 👍